java/src/test/java/io/cucumber/java/MethodScannerTest.java

```diff
@@ -29,10 +29,45 @@ void createBackend() {
     }
 
     @Test
-    void scan_finds_annotated_methods() throws NoSuchMethodException {
-        Method method = BaseSteps.class.getMethod("m");
+    void scan_finds_annotated_methods_in_public_class() throws NoSuchMethodException {
+        Method publicMethod = BaseSteps.class.getMethod("m");
+        Method packagePrivateMethod = BaseSteps.class.getDeclaredMethod("n");
+        Method protectedMethod = BaseSteps.class.getDeclaredMethod("o");
         MethodScanner.scan(BaseSteps.class, backend);
-        assertThat(scanResult, contains(new SimpleEntry<>(method, method.getAnnotations()[0])));
+        assertThat(scanResult,
+            contains(new SimpleEntry<>(publicMethod, publicMethod.getAnnotations()[0]),
```

Comment:
Consider using `org.hamcrest.collection.IsMapContaining.hasEntry` for all these.

Would have been better if I had done that the first time round, but better late then never.